actix-redis/src/session.rs

```diff
@@ -80,8 +80,9 @@ impl RedisSession {
     }
 
     /// Set custom cookie max-age
-    pub fn cookie_max_age(mut self, max_age: Duration) -> Self {
-        Rc::get_mut(&mut self.0).unwrap().max_age = Some(max_age);
+    /// Use `None` for session-only cookies
+    pub fn cookie_max_age(mut self, max_age: Option<Duration>) -> Self {
```

Comment:
This could be `max_age: impl Into<Option<Duration>>` for a bit nicer api.

Reply:
what can be converted into Duration?

Reply:
Not sure, but I mean for the `impl Into<Option<_>>`, so you can pass a value directly. Eh, it doesn't really matter.

Reply:
oh true, for the option<_> part that makes sense... yeah i prefer that

Reply:
not sure I follow, do you mean:
`pub fn cookie_max_age(mut self, max_age: impl Into<Option<_>>) -> Self {`

what does it do?

Reply:
there's a `impl From<T> for Option<T>` so that T's produce Some(T) and None is None.

Reply:
doing that would make this a nonbreaking change

Reply:
@cortopy It would let someone write:
```rust
cookie_max_age(duration); // we don't have to wrap it in `Some` because `T` implements `Into<Option<T>>`
cookie_max_age(Some(duration));
cookie_max_age((duration));
```

Reply:
Rust never ceases to impress me :) thanks the tip

However, I'm getting a compiler error:

```rust
pub fn cookie_max_age(mut self, max_age: impl Into<Option<_>>) -> Self {
        Rc::get_mut(&mut self.0).unwrap().max_age = max_age;
        self
    }
```

```
the type placeholder `_` is not allowed within types on item signatures
```

https://doc.rust-lang.org/error-index.html#E0121

Reply:
see first commetn

Reply:
I've pushed the changes but it's not my lucky night with CI. No clippy error but all tests pass in my machine, not sure what the error is about